run.py

```diff
@@ -440,6 +440,93 @@ def run_assembly_from_dir(asm_test_dir, iss_yaml, isa, mabi, gcc_opts, iss,
+    setting_dir : Generator setting directory
+    debug_cmd   : Produce the debug cmd log without running
+  """
+  if not c_test.endswith(".c"):
+    logging.error("%s is not a .c file" % c_test)
+    return
+  cwd = os.path.dirname(os.path.realpath(__file__))
+  c_test = os.path.expanduser(c_test)
+  report = ("%s/iss_regr.log" % output_dir).rstrip()
+  c = re.sub(r"^.*\/", "", c_test)
+  c = re.sub(r"\.c$", "", c)
+  prefix = ("%s/directed_c_tests/%s"  % (output_dir, c))
+  elf = prefix + ".o"
+  binary = prefix + ".bin"
+  iss_list = iss_opts.split(",")
+  run_cmd("mkdir -p %s/directed_c_tests" % output_dir)
+  logging.info("Compiling c test : %s" % c_test)
+
+  # gcc compilation
+  cmd = ("%s -mcmodel=medany -nostdlib \
+         -nostartfiles %s \
+         -I%s/user_extension \
+         -T%s/scripts/link.ld %s -o %s " % \
+         (get_env_var("RISCV_GCC", debug_cmd = debug_cmd), c_test, cwd,
+                      cwd, gcc_opts, elf))
+  cmd += (" -march=%s" % isa)
+  cmd += (" -mabi=%s" % mabi)
+  run_cmd_output(cmd.split(), debug_cmd = debug_cmd)
+  # Convert the ELF to plain binary, used in RTL sim
+  logging.info("Converting to %s" % binary)
+  cmd = ("%s -O binary %s %s" % (get_env_var("RISCV_OBJCOPY", debug_cmd = debug_cmd), elf, binary))
+  run_cmd_output(cmd.split(), debug_cmd = debug_cmd)
+  log_list = []
+  # ISS simulation
+  for iss in iss_list:
+    run_cmd("mkdir -p %s/%s_sim" % (output_dir, iss))
+    log = ("%s/%s_sim/%s.log" % (output_dir, iss, c))
+    log_list.append(log)
+    base_cmd = parse_iss_yaml(iss, iss_yaml, isa, setting_dir, debug_cmd)
+    logging.info("[%0s] Running ISS simulation: %s" % (iss, elf))
+    cmd = get_iss_cmd(base_cmd, elf, log)
+    run_cmd(cmd, 10, debug_cmd = debug_cmd)
+    logging.info("[%0s] Running ISS simulation: %s ...done" % (iss, elf))
+  if len(iss_list) == 2:
+    compare_iss_log(iss_list, log_list, report)
+
+def run_c_from_dir(asm_test_dir, iss_yaml, isa, mabi, gcc_opts, iss,
+                          output_dir, setting_dir, debug_cmd):
+  """Run a directed assembly test from a directory with spike
+
+  Args:
+    asm_test_dir    : Assembly test file directory
+    iss_yaml        : ISS configuration file in YAML format
+    isa             : ISA variant passed to the ISS
+    mabi            : MABI variant passed to GCC
+    gcc_opts        : User-defined options for GCC compilation
+    iss             : Instruction set simulators
+    output_dir      : Output directory of compiled test files
+    setting_dir     : Generator setting directory
+    debug_cmd       : Produce the debug cmd log without running
+  """
+  result = run_cmd("find %s -name \"*.c\"" % asm_test_dir)
+  if result:
+    asm_list = result.splitlines()
+    logging.info("Found %0d assembly tests under %s" %
+                 (len(asm_list), asm_test_dir))
+    for asm_file in asm_list:
+      run_assembly(asm_file, iss_yaml, isa, mabi, gcc_opts, iss, output_dir,
```

Comment:
should this be run_c?

Reply:
Oh I do not change it yet. Because `run_c` does not work. Could you check the options on gcc command?

Reply:
That's fine, we can have a follow up PR to make it work.